WooCommerce/Classes/ViewRelated/Coupons/CouponManagementListViewModel.swift

```diff
@@ -0,0 +1,170 @@
+import Foundation
+import Yosemite
+import protocol Storage.StorageManagerType
+import class AutomatticTracks.CrashLogging
+
+
+struct CouponListCellViewModel {
+    var title: String
+    var subtitle: String
+    var accessiblityLabel: String
+}
+
+enum CouponListState {
+    case initialized // Nothing to do here
```

Comment:
If we want to go get rid of it, it might be better to merge it with loading, because briefly showing an empty state first could be misleading...

`initialized` was added so that we could represent all the possible states of the screen: when we first init, we're not yet loading, we don't know whether there are coupons or not, etc. It's a habit from the RxSwift architecture we used for a while at my current job, definitely not something we need to keep here.

Reply:
I ended up bringing this back. Having an explicit change from `.initialized` to `.loading` gives the view controller an explicit `didEnter` state change after it tells the view model to start loading. This gives a straightforward way to start showing the placeholder cells, without the risk of showing the empty state briefly.